.github/release-checklist.md

```diff
@@ -0,0 +1,63 @@
+# Template to use for release PRs from `develop` to `main`
+
+:warning: **DO NOT MERGE (YET)** :warning:
+
+**Please **do** add approvals if you agree as otherwise we won't be able to release.**
+
+PR for tracking changes for the x.x.x release. Target release date: **DOW MONTH DAY YEAR**.
+
+## Release checklist
+
+### General
+
+- [ ] Verify, and if necessary, update the allowed version ranges for various dependencies in the `composer.json` - PR #xxx
+- [ ] PHPCS: check if there have been releases since the last WordPressCS release and check through the changelog to see if there is anything WordPressCS could take advantage of - PR #xxx
+- [ ] PHPCSUtils: check if there have been releases since the last WordPressCS release and update WordPressCS code to take advantage of any new utilities - PR #xxx
+- [ ] PHPCSExtra: check if there have been releases since the last WordPressCS release and check through the changelog to see if there is anything WordPressCS could take advantage of - PR #xxx
+- [ ] Check if the minimum WP version property needs updating in `MinimumWPVersionTrait::$default_minimum_wp_version` and if so, action it - PR #xxx
+- [ ] Check if any of the list based sniffs need updating and if so, action it.
+    List based sniffs:
+    - [ ] `WordPress.WP.ClassNameCase` - PR #xxx
+    - [ ] `WordPress.WP.DeprecatedClasses` - PR #xxx
+    - [ ] `WordPress.WP.DeprecatedFunctions` - PR #xxx
+    - [ ] `WordPress.WP.DeprecatedParameters` - PR #xxx
+    - [ ] `WordPress.WP.DeprecatedParameterValues` - PR #xxx
+- [ ] Check if any of the other lists containing information about WP Core need updating and if so, action it.
+    - [ ] `$allowed_core_constants` in `WordPress.NamingConventions.PrefixAllGlobals` - PR #xxx
+    - [ ] `$pluggable_functions` in `WordPress.NamingConventions.PrefixAllGlobals` - PR #xxx
+    - [ ] `$pluggable_classes` in `WordPress.NamingConventions.PrefixAllGlobals` - PR #xxx
+    - [ ] `$target_functions` in `WordPress.Security.PluginMenuSlug` - PR #xxx
+    - [ ] `$reserved_names` in `WordPress.NamingConventions.ValidPostTypeSlug` - PR #xxx
+    - [ ] `$wp_time_constants` in `WordPress.WP.CronInterval` - PR #xxx
+    - [ ] `$known_test_classes` in `IsUnitTestTrait` - PR #xxx
+    - [ ] ...etc...
+
+### Release prep
+
+- [ ] Add changelog for the release - PR #xxx
+    :pencil2: Remember to add a release link at the bottom!
+- [ ] Update `README` (if applicable) - PR #xxx.
+- [ ] Update wiki (new customizable properties etc) (if applicable)
+
+### Release
+
+- [ ] Merge this PR
+- [ ] Make sure all CI builds are green.
+- [ ] Tag and create a release against `main` (careful, GH defaults to `develop`!) & copy & paste the changelog to it.
+    :pencil2: Check if anything from the link collection at the bottom of the changelog needs to be copied in!
+- [ ] Make sure all CI builds are green.
```

Comment:
This is intentional? There is the same check on line 45

Reply:
Yes, CI builds run for the merge to `main` and then separate CI builds run for the tag. Builds for both events should be green, though it would be very exceptional if the outcome of these builds wouldn't match each other.

Still, for a release, I rather be sure.

Reply:
Ok, that makes sense. I thought that, since `main` merge would be ok, then the tag one should be as well. I mean it should be for all the checks, but could fail for external issues that should be checked for I guess 👍🏻

Reply:
Yes, and the builds which run for a push event (merge) don't have to be the same as the builds which run for a tag event, so better be sure.
(Example: in PHPCSUtils, a tag also triggers an update & deploy of the website, while a merge/push does not)